proposals/0000-uncurated-layer.rst

```diff
@@ -0,0 +1,57 @@
+3) Hackage will provide two package repository roots -- http://hackage.haskell.org and http://uncurated.hackage.haskell.org These roots will provide index-01.tar.gz files that contain the information, respectively, for curated packages, or for all packages. The uncurated root will contain no revision information for uncurated packages (even those that have been adopted), but will contain revision information for curated packages.
+4) Curated packages cannot depend on uncurated packages, and the hackage server will detect this as an error at upload time.
+5) Uncurated packages may be "adopted" into the curated ecosystem by trustees. Metadata revisions necessarily remove the x-uncurated property from the revised cabal metadata.
+
+The first phase of this rollout is simply social. It has the following properties:
+
++ Hackage trustees will recognize and respect the uncurated flag. When it is set to "uncurated-no-trustee-contact" they will not contact authors with any issues. They *will* retain the ability to make metadata revisions, bearing in mind that they must remove the x-uncurated property from revised metadata.
+
+The second phase is a technical change as soon as possible to enforce the semantics of x-curation:
+
++ Hackage will ensure that no revision has x-curation set to any variant of "uncurated".
+
+The third phase is implementation of UI:
+
++ The curation flag will be detected and displayed on a package's page, as part of the general data provided about a package. It will also be provided in search and browse results. Ideally, search and browse results will be extended in general with the ability to perform in-page filtering on flags and fields, such as "library", curated status, deprecation status, perhaps presence in distributions or compatibility with ghc versions, etc. (note: integration with tags may be a mechanism for this).
+
+The fourth phase is indices:
+
++ The uncurated package repo root will be built and provided.
+
+Fifth, Hackage can now begin to gauge the costs of enforcing the policy regarding curated packages not depending on uncurated packages (or, more precisely, having an install-plan that can operate purely out of the curated index).
+
++ Warnings will be given in such situations, and statistics will be collected as to the frequency of these warnings and the blocking packages in need of adoption.
+
+Sixth, when there is confidence among the trustees that the impact will not be too significant, the policy can be enforced.
+
++ Curated package uploads will be checked on upload to ensure they don't have dependencies on uncurated packages. Further, the curated index will only provide information on curated packages.
+
+At this point we can proceed to a discussion on default filtering settings, bearing in mind that settings can always be saved and persisted per-user.
```

Comment:
I am commenting here again because of [this comment on Reddit](https://www.reddit.com/r/haskell/comments/7yfdei/haskell_ecosystem_requests/dugl11i/):

> No there's been no comment on the thread after over a week, which was the stated cutoff for "I guess everyone's ok with this then." So it is ready to implement imho.

I am not okay with the proposal as it is now, and taking my silence as tacit approval feels weird. As [I stated earlier](https://github.com/haskell/ecosystem-proposals/pull/6#issuecomment-360952030), I strongly disagree with hiding uncurated packages from the UI entirely. Is that decision part of this discussion, or is it going to happen at some later time?